### PR TITLE
Javacc2

### PR DESCRIPTION
**Pull request by wafman**
- [ ] Top-level README “Table of Contents” is updated
 - [x ] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
     - [x ] Binary Search - Expected outcome
     - [ x] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [ ] README for this challenge is complete
     - [ ] Summary, Description, Approach & Efficiency, Solution
     - [ ] Link to code
     - [ ] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/main/java/code401Challenges/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        int low = 0;
+        int high = arr.length - 1;
+        while(low <= high){
+            int mid = (low + high) / 2;
+            if(arr[mid] < num){
+                low = mid + 1;
+            } else if(arr[mid] > num){
+                high = mid - 1;
+            } else if(arr[mid] == num) {
+                index = mid;
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/BinarySearch.java]: If `arr[mid] == num` is true couldn't you just return `mid`?

File: code401Challenges/src/main/java/code401Challenges/BinarySearch.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                break;
+            }else if(mid != num){
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/BinarySearch.java]: Either arr[mid] is less than num, is more than num, or is equal to num. Since you have all those possibilities covered I don't think this final `else if` is reachable
